File: problems/dfs_and_bfs/PR_43163_단어변환.py
from collections import deque

def solution(begin, target, words):
    if target not in words:
        return 0
    
    return bfs(begin, words, target)


def bfs(begin, words, target):
    need_visit = deque([[begin, 0]])

    while need_visit:
        begin, count = need_visit.popleft()

        if begin == target:
            return count

        count += 1
        
        for i in range(len(words)):
            if comp_str(begin, words[i]) == 1:
                need_visit.append([words[i], count])

    return 0

def comp_str(str1, str2):
    count = 0
    for a, b in zip(str1, str2):
        if a != b:
            count += 1
    
    return count

File: problems/dfs_and_bfs/PR_43163_단어변환.js
function solution(begin, target, words) {
  if (!words.includes(target)) return 0
  const need_visit = [[begin, 0]]
  while (need_visit) {
    const [word, count] = need_visit.shift()

    if(word === target) {
      return count
    }

    words.forEach(next => {
      if (check(word, next))
        need_visit.push([next, count+1])
    })
  }
  
  var answer = 0;
  return answer;
}

function check(str1, str2) {
  const a = str1.split("")
  const b = str2.split("")
  let count = 0
  a.forEach((char, i) => {
      if(char !== b[i])
          count++
  })
  if (count===1)
      return true
  else
      return false
}

console.log(solution("hit", "cog", ["dog", "hot", "dot", "cog", "lot", "log"]))
console.log(check("hit", "hop"))